src/contexts/DpiTokenMarketData/DpiTokenMarketDataProvider.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useEffect } from 'react'
+import MarketDataContext from './DpiTokenMarketDataContext'
+import { coingeckoDpiId } from 'constants/coingeckoIds'
+import { fetchHistoricalTokenMarketData } from 'utils/coingeckoApi'
+
+interface HistoricalMarketDataProvider {
+  // coingeckoId: string
+	daysOfHistory: number
+}
+
+const DpiMarketDataProvider: React.FC<HistoricalMarketDataProvider> = ({
+  children,
+  daysOfHistory,
+}) => {
+  const [dpiMarketData, setDpiMarketData] = useState<any>({})
+  useEffect(() => {
+    const endTime = Date.now() / 1000;
+    const startTime = endTime - (86400 * daysOfHistory);
+    fetchHistoricalTokenMarketData(coingeckoDpiId, startTime, endTime)
+      .then((response: any) => {
+        setDpiMarketData(response)
+      })
+      .catch((error: any) => console.log(error));
+  }, [])
+
+  const getLatest = (arr?: number[][]) =>
+    arr && arr.length ? arr[arr.length - 1][1] : 0
```

Comment:
Label these a bit more clearly. `arr` could be `tokenMarketData`? and `getLatest` could be `selectLatestMarketData`?

Also consider using optional chaining here

Reply:
For function name I'l definitely change,  called it `getLatest()` because it's basically just a double `last()`.
For the function body optional chaining doesn't make the code that much shorter and for me, my original code is easier to read but I've never used optional chaining before so that could just be me
`marketData?.[marketData.length -1]?.[1] || 0` vs `arr && arr.length ? arr[arr.length - 1][1] : 0`

Reply:
Ok let's keep it in your original style then, just with updated variable names.

Reply:
Oh it's actually uncomfortably long once the variable name is changed, I'll just use the optional chaining lol